include/cantera/numerics/PreconditionerBase.h

```diff
@@ -0,0 +1,158 @@
+/**
+ *  @file PreconditionerBase.h Declarations for the class
+ *   PreconditionerBase which is a virtual base class for
+ *   preconditioning systems.
+ */
+
+// This file is part of Cantera. See License.txt in the top-level
+// directory or at https://cantera.org/license.txt for license and
+// copyright information.
+
+#ifndef PRECONDITIONERBASE_H
+#define PRECONDITIONERBASE_H
+
+#include "cantera/base/ctexceptions.h"
+#include "cantera/numerics/Integrator.h"
+
+namespace Cantera
+{
+
+//! Forward declarations
+class ReactorNet;
+class Reactor;
+class MoleReactor;
+class IdealGasMoleReactor;
+class IdealGasConstPressureMoleReactor;
+
+//! Flag to indicate preconditioner is not set
+const int PRECONDITIONER_NOT_SET = 0;
+
+//! PreconditionerBase serves as an abstract type to extend different
+//! preconditioners from.
+class PreconditionerBase
+{
+public:
+    PreconditionerBase(/* args */){}
+    ~PreconditionerBase(){}
+
+    //! Use this function to return zero for preconditioner not set
+    virtual size_t getPreconditionerMethod(){return PRECONDITIONER_NOT_SET;};
+
+    //! Use this function to the "type" for CVODES
+    virtual PreconditionerType getPreconditionerType(){return NO_PRECONDITION;};
```

Comment:
Note the following locations for whitespace generally, and that the `;` is not necessary after the body of a function.
```suggestion
    virtual PreconditionerType getPreconditionerType() { return NO_PRECONDITION; }
```